crates/wasmtime/src/runtime/profiling.rs

```diff
@@ -125,6 +156,26 @@ impl GuestProfiler {
         }
     }
 
+    /// Create a new profiler for the provided component
+    ///
+    /// See [`GuestProfiler::new`] for additional information; this function
+    /// works identically except that it takes a component and sets up
+    /// instrumentation to track calls in each of its constituent modules.
+    #[cfg(feature = "component-model")]
+    pub fn new_component(
+        component_name: &str,
+        interval: Duration,
+        component: Component,
+        extra_modules: impl IntoIterator<Item = (String, Module)>,
+    ) -> Self {
```

Comment:
It probably makes sense to use a builder-style API in the limit here, to allow users to specify all the various components and modules they are profiling, but this is totally fine for the moment.